### PR TITLE
Feature/fetch events on category page load

### PR DESCRIPTION
**Pull request by cgolinski**
Advice on which sections could have been split out to make this a smaller PR very welcome!

Styling will need to be fixed. 'Set your location' button is now unclickable under the header.

### CODE REVIEW

File: src/redux/epics/event.epics.js [hunk 1]
```diff
@@ -9,7 +9,10 @@ export const getEventEpic = (action$, store) =>
   action$.ofType(ACTION_TYPES.GET_EVENTS)
     .mergeMap(action =>
       Observable.ajax({
-        url: `${BASE_ENDPOINT}${store.getState().category.chosenCategory.id}`,
+        url: `${BASE_ENDPOINT}${
```
Comment (epq) [inline src/redux/epics/event.epics.js]: With category id, url becomes:
```
  url: `${BASE_ENDPOINT}${action.payload.categoryID}`,
```

File: src/components/CategoryPage.js [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-    </button>
-    {props.events 
-      ? <EventsList events={props.events} />
-      : null
-    }
-  </div>
-); 
+export class CategoryPage extends Component {
+  componentDidMount() {
+    this.props.getEvents();
```
Comment (epq) [inline src/components/CategoryPage.js]: Alternatively, pass the category id:
```
componentDidMount() {
    this.props.getEvents({...this.props, categoryID: this.props.params.categoryID});
  }
```

File: src/components/CategoryPage.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+    return (
+      <div className="category-page">
+        {this.props.allCategories.map(category => category.id === this.props.params.categoryID 
+          ? <h2>{category.name} Events near Toronto, Ontario</h2> 
+          : null)
+        }
+        {this.props.events 
+          ? <EventsList events={this.props.events} />
+          : null
+        }
```
Comment (auredufour) [inline src/components/CategoryPage.js]: If not, we should return an error message or something saying there is no result

File: src/components/CategoryPage.js [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+  componentDidMount() {
+    this.props.getEvents();
+  }
+
+  render() {
+    return (
+      <div className="category-page">
+        {this.props.allCategories.map(category => category.id === this.props.params.categoryID 
+          ? <h2>{category.name} Events near Toronto, Ontario</h2> 
+          : null)
```
Comment (auredufour) [inline src/components/CategoryPage.js]: I have an error when I try to run the code, it says that 'name" is null.
It only happen if I refresh the page, not when I go to it!

Comment (cgolinski) [inline src/components/CategoryPage.js]: I can't seem to reproduce this error, but am about to push a change, so try again in a couple minutes and see if it's still occurring.

File: src/components/CategoryPage.js (hunk 3 above)
Comment (cgolinski) [inline src/components/CategoryPage.js]: Let's do this as part of a different PR.

**Comment by auredufour** [comment]
👍

**Comment by cgolinski** [comment]
Updated to fetch events on location_change to /category/:categoryID. Added regex to identify route in event epic.

File: src/redux/epics/event.epics.js [hunk 5]
```diff
@@ -4,12 +4,13 @@ import { Observable } from 'rxjs';
 import 'rxjs/add/operator/mergeMap';
 
 const BASE_ENDPOINT = `https://www.eventbriteapi.com/v3/events/search/?token=${EVENTBRITE_API_KEY}&location.address=toronto&categories=`;
+const reCategoryPath = /^.*\/category\/\d\d\d$/;
 
 export const returnEventActionOnLocationChange = (action$) =>
```
Comment (cgolinski) [inline src/redux/epics/event.epics.js]: I thought about something similar, but this epic doesn't make the api call or get the events, it just returns the getEvents action.

File: src/redux/epics/event.epics.js [hunk 6]
```diff
@@ -5,14 +5,19 @@ import 'rxjs/add/operator/mergeMap';
 
 const BASE_ENDPOINT = `https://www.eventbriteapi.com/v3/events/search/?token=${EVENTBRITE_API_KEY}&location.address=toronto&categories=`;
 
-export const getEventEpic = (action$, store) =>
+export const returnEventActionOnLocationChange = (action$) =>
+  action$.ofType('@@router/LOCATION_CHANGE')
+    .filter(action => action.payload.pathname.includes('category/'))
```
Comment (cgolinski) [inline src/redux/epics/event.epics.js]: Good catch. I'll add that.

Comment (cgolinski) [inline src/redux/epics/event.epics.js]: I think you might be reviewing an old version of the code. I have already updated this to:
```
const reCategoryPath = /^.*\/category\/\d\d\d$/;

export const returnEventActionOnLocationChange = (action$) =>
  action$.ofType('@@router/LOCATION_CHANGE')
    .filter(action => reCategoryPath.test(action.payload.pathname))
```

File: src/redux/epics/event.epics.js (hunk 5 above)
Comment (turquoisemelon) [inline src/redux/epics/event.epics.js]: Oh you're right. I just realized the 'getEventEpic ' below this

File: src/redux/epics/event.epics.js [hunk 7]
```diff
@@ -4,12 +4,21 @@ import { Observable } from 'rxjs';
 import 'rxjs/add/operator/mergeMap';
 
 const BASE_ENDPOINT = `https://www.eventbriteapi.com/v3/events/search/?token=${EVENTBRITE_API_KEY}&location.address=toronto&categories=`;
+const reCategoryPath = /^.*\/category\/\d\d\d$/;
```
Comment (epq) [inline src/redux/epics/event.epics.js]: Are category ids always three digit numbers? It seems like they are, but you could replace it with this to be more flexible to match one or more numbers:
```
const reCategoryPath = /^\/category\/\d\d\d$/;
```
Is there a reason why you are matching `.*` at the beginning? The pathname is going to start with `/category` so you could leave this out.

Comment (epq) [inline src/redux/epics/event.epics.js]: I didn't paste the proper regex above; it's fixed now.

**Comment by srhboo** [comment]
wow nice job! 👍 the slightly more flexible regex expression that @epq suggested could be good to change to - but if cat IDs are 3 digits, looks good to go!
